src/checkout.ts

```diff
@@ -94,3 +104,24 @@ function getOrCreateCheckoutElement() {
 
 	return element as PurpleDotCheckoutElement;
 }
+
+async function cartRequiresSeparateCheckout(cartItems: CartItem[]) {
+	const results = await Promise.all(
+		cartItems.map((cartItem) => cartItemRequiresSeparateCheckout(cartItem)),
+	);
+	return results.some((result) => result);
+}
```

Comment:
This is potentially going to do quite a lot of requests at once. Did you give up on the addind attributes to the actual cart like shopify-script does idea?

Reply:
I think there is a possibility here that the call we do to `fetchVariantsPreorderState` here ends up returning different waitlists from the one the item was originally assigned to which means you'd end up in the wrong checkout based on the waitlists?

Reply:
That's not really the main purpose of fixCart, it's main goal is to make sure things get release IDs (or not) as needed.